dotnet-local.cmd

```diff
@@ -1,8 +1,9 @@
 @echo off
+setlocal enabledelayedexpansion
 SET ROOT=%~dp0
 IF EXIST "%ROOT%\bin\dotnet\dotnet.exe" (
-    SET DOTNET_ROOT=%ROOT%\bin\dotnet
-    SET PATH=%DOTNET_ROOT%;%PATH%
+    SET "DOTNET_ROOT=%ROOT%\bin\dotnet"
+    SET "PATH=!DOTNET_ROOT!;%PATH%"
```

Comment:
@vitek-karas do we have any similar problems here?

https://github.com/dotnet/android/blob/main/dotnet-local.cmd

Or is it fine? Because there are not multiple lines inside an `IF`?

Reply:
That should be fine - the problem is that the parser considers the entire IF as one statement and performs env. variable replacement BEFORE it executes the statement. Any changes to the environment are not visible to the statements inside the IF, but they are visible to external tools (so in this case the `dotnet.exe` will see the new values as expected).